src/core/thread/mesh_forwarder.cpp

```diff
@@ -572,81 +572,84 @@ Message *MeshForwarder::PrepareNextDirectTransmission(void)
 {
     Message *curMessage, *nextMessage;
     Error    error = kErrorNone;
-
-    for (curMessage = mSendQueue.GetHead(); curMessage; curMessage = nextMessage)
-    {
-        // We set the `nextMessage` here but it can be updated again
-        // after the `switch(message.GetType())` since it may be
-        // evicted during message processing (e.g., from the call to
-        // `UpdateIp6Route()` due to Address Solicit).
-
-        nextMessage = curMessage->GetNext();
-
-        if (!curMessage->IsDirectTransmission() || curMessage->IsResolvingAddress())
-        {
-            continue;
-        }
-
+    
 #if OPENTHREAD_CONFIG_DELAY_AWARE_QUEUE_MANAGEMENT_ENABLE
-        if (UpdateEcnOrDrop(*curMessage, /* aPreparingToSend */ true) == kErrorDrop)
-        {
-            continue;
-        }
+    IgnoreError(RemoveAgedMessages());
 #endif
-        curMessage->SetDoNotEvict(true);
 
-        switch (curMessage->GetType())
+    for (uint8_t priority = Message::kPriorityNet; priority >= Message::kPriorityLow; priority--)
```

Comment:
`SendQueue` is a `PriorityQueue`, so the messages in it are already stored based on their priority, no need to do two `for` loops here.

Reply:
> Account for message priority while prepare and send direct transmission

Should be already accounted due to use of `PriorityQueue`.